### PR TITLE
Distinguish between message type for lookup fields

### PR DESCRIPTION
**Pull request by chukarave**
for each lookup field, ascertain which type of message key
is required (field empty/entry invalid) and use that while committing
This is a chunk broken down from [PR #223](https://github.com/wmde/new-lexeme-special-page/pull/223)

Bug: T310134

### CODE REVIEW

**Review by micgro42 — APPROVED** [review]

**Comment by lucaswerkmeister** [comment]
Can we add a test for this as well?

**Comment by micgro42** [comment]
yeah, just noticed that after I clicked approve

**Review by lucaswerkmeister — DISMISSED** [review]
LGTM; I wonder if we should also add the search inputs (as empty string) to the other test for `null` language / lexical category?

```diff
diff --git a/tests/unit/store/actions.test.ts b/tests/unit/store/actions.test.ts
index 74acf355e4..13adf5d835 100644
--- a/tests/unit/store/actions.test.ts
+++ b/tests/unit/store/actions.test.ts
@@ -196,7 +196,9 @@ describe( CREATE_LEXEME, () => {
 				return {
 					lemma: 'example lemma',
 					language: null,
+					languageSearchInput: '',
 					lexicalCategory: null,
+					lexicalCategorySearchInput: '',
 					spellingVariant: 'en',
 				} as RootState;
 			},
```

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**
